components/base/wheeled/kinematics.go

```diff
@@ -51,13 +61,74 @@ func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
 }
 
 func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
-	// TODO(RSDK-2311): complete the implementation
-	return []referenceframe.Input{}, errors.New("not implemented yet")
+	// TODO(rb): make a transformation from the component reference to the base frame
+	pose, _, err := kwb.slam.GetPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pt := pose.Point()
+	theta := math.Mod(pose.Orientation().OrientationVectorRadians().Theta, 2*math.Pi) - math.Pi
+	return []referenceframe.Input{{Value: pt.X}, {Value: pt.Y}, {Value: theta}}, nil
 }
 
-func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	// TODO(RSDK-2311): complete the implementation
-	return errors.New("not implemented yet")
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, inputs []referenceframe.Input) (err error) {
+	// create a frame system to be used during the context of this function, used to locate the goal in the base frame
+	fs := referenceframe.NewEmptyFrameSystem("")
+	if err := fs.AddFrame(kwb.model, fs.World()); err != nil {
+		return err
+	}
+
+	// create a function for the error state, which is defined as [positional error, heading error]
+	errorState := func() (int, float64, error) {
+		currentInputs, err := kwb.CurrentInputs(ctx)
+		if err != nil {
+			return 0, 0, err
```

Comment:
There is no equivalent to NaN for ints; 0 or -1 is the best you can do.

Since it's against style to return a non-nil error alongside any meaningful value; this *should* be properly handled by the caller.

Reply:
int = 0 it is then. I always like the clarity afforded by `math.Nan`.